src/OpenCensus.Collector.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -0,0 +1,161 @@
+        private readonly PropertyFetcher stopExceptionFetcher = new PropertyFetcher("Exception");
+        private readonly PropertyFetcher stopRequestStatusFetcher = new PropertyFetcher("RequestTaskStatus");
+
+        public HttpHandlerDiagnosticListener(ITracer tracer, ISampler sampler) : base("HttpHandlerDiagnosticListener", tracer, sampler)
+        {
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+            if (!(this.startRequestFetcher.Fetch(payload) is HttpRequestMessage request))
+            {
+                // Debug.WriteLine("request is null");
+                return;
+            }
+
+            // TODO: this needs to be generilized
+            if (request.RequestUri.ToString().Contains("zipkin.azurewebsites.net"))
+            {
+                return;
+            }
+
+            this.LocalScope.Value = this.Tracer.SpanBuilder("HttpOut").SetSampler(this.Sampler).StartScopedSpan();
+            var span = this.Tracer.CurrentSpan;
+            span.PutClientSpanKindAttribute();
+            span.PutHttpMethodAttribute(request.Method.ToString());
+            span.PutHttpHostAttribute(request.RequestUri.Host);
+            span.PutHttpPathAttribute(request.RequestUri.AbsolutePath);
+            span.PutHttpUrlAttribute(request.RequestUri.ToString());
+        }
+
+        public override void OnStopActivity(Activity activity, object payload)
+        {
+            if (!(this.stopResponseFetcher.Fetch(payload) is HttpResponseMessage response))
+            {
+                // Debug.WriteLine("response is null");
+                return;
+            }
+
+            var requestTaskStatus = this.stopRequestStatusFetcher.Fetch(payload) as TaskStatus?;
+
+            var span = this.Tracer.CurrentSpan;
+            if (requestTaskStatus.HasValue)
+            {
+                if (requestTaskStatus != TaskStatus.RanToCompletion)
+                {
+                    span.PutErrorAttribute(requestTaskStatus.ToString());
+                }
+            }
+
+            if ((int)response.StatusCode >= 200 && (int)response.StatusCode <= 301)
+            {
+                span.Status = Status.Ok;
+            }
+            else if ((int)response.StatusCode == 401)
+            {
+                span.Status = Status.Unauthenticated;
+            }
+            else if ((int)response.StatusCode == 403)
+            {
+                span.Status = Status.PermissionDenied;
+            }
+            else if ((int)response.StatusCode == 404)
+            {
+                span.Status = Status.NotFound;
+            }
+            else if ((int)response.StatusCode == 501)
+            {
+                span.Status = Status.Unimplemented;
+            }
+            else
+            {
+                span.Status = Status.Unknown;
+            }
+
+            span.Status = span.Status.WithDescription(response.StatusCode + " " + response.ReasonPhrase);
```

Comment:
assuming task has faulted and there is no response, we should add exception details here (in onException handler)

Reply:
definitely